app/scripts/directives/notifications/notificationCounter.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+(function() {
+
+  angular
+    .module('openshiftConsole')
+    .component('notificationCounter', {
+      templateUrl: 'views/directives/notifications/notification-counter.html',
+      bindings: {},
+      controller: [
+        '$filter',
+        '$routeParams',
+        '$rootScope',
+        'AuthService',
+        'Constants',
+        NotificationCounter
+      ]
+    });
+
+  function NotificationCounter($filter, $routeParams, $rootScope, AuthService, Constants) {
+
+      var counter = this;
+      var DISABLE_GLOBAL_EVENT_WATCH = _.get(Constants, 'DISABLE_GLOBAL_EVENT_WATCH');
+      var INSUFFICIENT_WATCHES = $filter('isIE')() || $filter('isEdge')();
+
+      counter.hide = true;
+
+      var rootScopeWatches = [];
+      // this one is treated separately from the rootScopeWatches as
+      // it may need to be updated outside of the lifecycle of init/destroy
+      var notificationListeners = [];
+
+      var watchNotificationDrawerCount = function(projectName, cb) {
+        if(!projectName) {
+          return;
+        }
+        notificationListeners.push($rootScope.$on('NotificationDrawerWrapper.count', cb));
+      };
+
+      var deregisterNotificationListeners = function() {
+        _.each(notificationListeners, function(listener) {
+          listener && listener();
+        });
+        notificationListeners = [];
+      };
+
+      var deregisterRootScopeWatches = function() {
+        _.each(rootScopeWatches, function(deregister) {
+          deregister();
+        });
+        rootScopeWatches = [];
+      };
+
+      var hideIfNoProject = function(projectName) {
+        if(!projectName) {
+          counter.hide = true;
+        } else {
+          counter.hide = false;
+        }
+      };
+
+      counter.onClick = function() {
+        $rootScope.$emit('NotificationDrawerWrapper.toggle');
+      };
+
+      var drawerCountCallback = function(event, newCount) {
+        if(newCount) {
+          counter.showNewNotificationIndicator = true;
+        } else {
+          counter.showNewNotificationIndicator = false;
+        }
+      };
+
+      var projectChanged = function(next, current) {
+        return _.get(next, 'params.project') !== _.get(current, 'params.project');
+      };
+
+      var reset = function() {
+        watchNotificationDrawerCount($routeParams.project, drawerCountCallback);
+        hideIfNoProject($routeParams.project);
+      };
+
+      counter.$onInit = function() {
+        if(DISABLE_GLOBAL_EVENT_WATCH || INSUFFICIENT_WATCHES || !AuthService.isLoggedIn()) {
```

Comment:
I wonder if the timing is different because you're using `$onInit`.

If we disable when `!AuthService.isLoggedIn()`, it's good to add `AuthService.onUserChanged(function() { ... })` to start the watch.

nit: I suggest using the same terminology `LIMIT_WATCHES` as we use on the overview (and a couple other pages) so it's clear this variable is essentially the same.